src/oop/special_class_method/hard/q14/MainTest.java

```diff
@@ -0,0 +1,167 @@
+package oop.special_class_method.hard.q14;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Parsons with Distractors
+
+    private final String className = "Bottle";
+    private final String attributeName = "isFull";
+    private final String getAttributeMethodName = "getIsFull";
+    private final String setAttributeMethodName = "setIsFull";
+    private ObjectTest testClass;
+
+    @BeforeEach
+    public void setUp() throws Throwable {
+        String classString = "oop.special_class_method.hard.q14." + className;
+        this.testClass = new ObjectTest(classString);
+    }
+
+    @Test
+    public void bottleClassHasCorrectAttributes() {
+        assertTrue(testClass.hasField(attributeName),
+                "Your " + className + " class is missing the " + attributeName + " attribute, or it is spelled incorrectly.");
+        assertTrue(testClass.hasField(attributeName, boolean.class),
+                "Your " + attributeName + " does not have the correct type.");
+        assertTrue(testClass.hasModifier(attributeName, "private"),
+                "Your " + attributeName + " does not have the correct visibility modifier.");
+    }
+
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(true, false),
+                Arguments.of(false, true),
+                Arguments.of(true, true),
+                Arguments.of(false, false)
+        );
+    }
+
+    @Test
+    public void bottleClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {};
+        assertTrue(testClass.hasConstructor(classArguments),
+                "Your " + className + " constructor does not have the correct parameters.");
+        assertTrue(testClass.hasModifier(classArguments, "public"),
+                "Your " + className + " constructor does not have the correct visibility modifiers.");
+    }
+
+    @Test
+    public void bottleClassHasCorrectConstructor() throws Throwable {
```

Comment:
I would do a repeated test here, just run it 3 or 5 times to make sure it is always false.